cmd/dra-example-kubeletplugin/checkpoint.go

```diff
@@ -3,14 +3,19 @@ package main
 import (
 	"encoding/json"
 
+	"k8s.io/kubernetes/pkg/kubelet/checkpointmanager"
```

Comment:
`k8s.io/kubernetes` should not be imported. Any code inside is considered internal and not meant for public consumption. There are some exceptions (most notably the scheduler framework for building custom schedulers), but not this one here.

It's not even a particularly good package. We had huge issues with figuring out how checksumming was meant to be used and what the purpose of checksumming was in the first place.

Can we perhaps use this opportunity to drop the dependency?

Reply:
cc @pohly 
> Can we perhaps use this opportunity to drop the dependency?

=> In terms of dependency, I'm on the same page. I will update PR to introduce simple checkpoint util to drop the dependency.
But for checksum do you mean we don't need checksum here?  Seems there was some issues with dra_manager_state in the past. Or do you mean we need well designed checkpoint implementation along with checksum. Since it is example dra driver, maybe simple checkpointing without checksum is fine.

Reply:
The only purpose of checksumming that I could imagine is to detect bit flips in the file. As a DRA driver author, is that important to you on top of whatever potential checksumming and error correcting the OS might do?

Are there other reasons for it?

Reply:
I haven’t thought about it seriously.
The checkpoint manager already existed when I was involved with resource managers in kubelet around 2019.

Seems, checksum is originated from PodSandbox checkpointer of dockershim.
- https://github.com/kubernetes/kubernetes/commit/d62bd9ef6571c13b134162da2f7ae41785c6a59e#diff-7cf7a43176686c06054024ad9d7226062787c188c66fd17d837d7e821b164ab0

Approaching conservatively, there seem to be a few possible cases:
- The process could restart while writing a file (e.g., OOM, kill, restart).
- The file might be corrupted unintentionally by a person or script.

Reply:
> The process could restart while writing a file (e.g., OOM, kill, restart).

The usual approach is to write a temp file, sync, then rename. But I suppose a checksum is easier.

> The file might be corrupted unintentionally by a person or script.

True, albeit a bit unlikely.

Reply:
> I will update PR to introduce simple checkpoint util to drop the dependency.

Gentle reminder that this is pending.

/lgtm cancel

Reply:
I'm fine without a checkpoint checksum. For the "writing file fails" case I think the "write tmp file, sync, close, remove, rename" approach would be useful. I don't know about other existing mechanisms that could be used here.

cc @nojnhuh

Reply:
Yep, I'll pick this up.

/remove-lifecycle rotten